transport-native-unix-common/src/main/c/netty_unix_socket.c

```diff
@@ -183,6 +189,27 @@ static int socket_type(JNIEnv* env) {
         }
         return AF_INET6;
     } else {
+        int optval = 1;
+        if (netty_unix_socket_setOption0(fd, IPPROTO_IPV6, IPV6_V6ONLY, &optval, sizeof(optval)) < 0) {
+            struct sockaddr_in6 ip6addr;
+
+            ip6addr.sin6_family = AF_INET6;
```

Comment:
I'm not a C pro, but would this be clearer as:

```
memset(&ip6addr, 0, sizeof(struct sockaddr_in6))
ip6addr.sin6_family = AF_INET6;
ip6addr.sin6_addr = in6addr_any;
```

It's a few lines shorter, and shows you didn't miss any fields.  Thoughts?